addon-test-support/scroll.js

```diff
@@ -0,0 +1,47 @@
+import getElementWithAssert from './-private/get-element-with-assert';
+import wait from 'ember-test-helpers/wait';
+
+
+/*
+  Triggers a paint (and therefore flushes any queued up scroll events).
+
+  @method triggerFlushWithPromise
+  @return {RSVP.Promise}
+  @private
+*/
+function triggerFlushWithPromise(){
+  return wait().then(() => {
```

Comment:
I'm not familiar with the problem, but intuitively I feel that the order should be the opposite. My understanding is that the scroll event is not fired until the next time the screen is painted. As a consequence of that scroll event, some javascript will be executed and maybe some asynchronous behaviour will be triggered, and we should wait for it.

It seems to me that first we should wait for the `requestAnimationFrame` so the event is fired and then `wait` for whatever work that event handler started.